docs/001_develop/03_client-capabilities/008_interaction/switch.mdx

```diff
@@ -0,0 +1,149 @@
+---
+title: 'Switch'
+sidebar_label: 'Switch'
+id: client-interaction-switch
+keywords: [interaction, interactive, switch, form]
+tags:
+- interaction
+- interactive
+- switch
+- form
+
+sidebar_position: 8
+---
+import SwitchDemo from '../../../../examples/ui/client-capabilities/interaction/switch.js';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Switch
+
+A form-connected implementation of a [switch](https://w3c.github.io/aria/#switch) component.
+
+Use cases:
+* Toggling visibility of a UI section
+* Disabling & enabling features
+
+## Example
+
+<SwitchDemo />
+
+<Tabs defaultValue="genesis" values={[{ label: 'Genesis', value: 'genesis', }, { label: 'React', value: 'react', }, { label: 'Angular', value: 'angular', }]}>
+
+<TabItem value="genesis">
+
+Declaration
+```html
+<rapid-switch checked="${checked}"></rapid-switch>
+```
+
+Usage
+```typescript
+@customElement({
+  name: 'my-element',
+  template: html`
+    <rapid-switch>
+      Theme
+      <span slot="checked-message">Dark</span>
+      <span slot="unchecked-message">Light</span>
+    </rapid-switch>
+  `,
+})
+export class MyElement extends GenesisElement {
+}
+```
+
+</TabItem>
+<TabItem value="react">
+
+Declaration
+```html
+<rapid-switch></rapid-switch>
+```
+Usage
+```typescript
+export function MyComponent() {
+  const [switchValue, setSwitchState] = useState(false);
+  const handleSwitchChange = (event) => {
+    setSwitchState(!!event.target.checked);
+    console.log({ switchValue })
+  };
+
+  return (
+    <rapid-switch onChange={handleSwitchChange}></rapid-switch>
+  );
+}
+```
+</TabItem>
+<TabItem value="angular">
+
+Declaration
+```html
+<rapid-switch></rapid-switch>
+```
+
+Usage
+```typescript
+import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+@Component({
+  selector: 'my-root',
+  template: `
+    <rapid-switch
+      name="isSelected"
+      [checked]="isSelected"
+      (change)="onSwitchChange($event)"
```

Comment:
same here -  we don't have it in the foundation example

Reply:
OK I added to the genesis code example.